File: 백준/Silver/11292. 키 큰 사람/키 큰 사람.py
N = int(input())

while N!=0:
    test_case = []
    for i in range(N):
        name, height = input().split()
        height = float(height)
        test_case.append((name, height))
    
    test_case.sort(key=lambda x: (x[1]), reverse=True)
    max_height = test_case[0][1]
    
    for name, height in test_case:
        if height==max_height:
            print(name)
        else:
            break
    
    N = int(input())

File: 백준/Silver/11292. 키 큰 사람/키 큰 사람.c
#include <stdio.h>

int main(){
    char name[100][10];
    float height[100];
    
    int N;
    
    scanf("%d", &N);
    
    while (N>0) {
        for (int i=0; i<N; i++){
            scanf("%s %f", &name[i], &height[i]);
        }
        float max_height = 0;
        for (int i=0; i<N; i++){
            if (height[i] > max_height){
                max_height = height[i];
            }
        }
        for (int i=0; i<N; i++){
            if (height[i]==max_height) {
                printf("%s\n", name[i]);
            }
        }
        scanf("%d", &N);       
    }
  
    return 0;
}